astroid/nodes/scoped_nodes/mixin.py

```diff
@@ -170,3 +173,19 @@ class ComprehensionScope(LocalsDictNodeNG):
     """Scoping for different types of comprehensions."""
 
     scope_lookup = LocalsDictNodeNG._scope_lookup
+
+    generators: List["nodes.Comprehension"]
+    """The generators that are looped through."""
+
+    def qname(self):
```

Comment:
This is another instance of https://github.com/pycqa/astroid/issues/278. We could also change `qname()` on `LocalsDictNodeNG` not to use `name()` if it doesn't exist.

I don't know what is better.

Reply:
```suggestion
    def qname(self) -> str:
```